packages/sdk/src/Trackable.ts

```diff
@@ -0,0 +1,49 @@
+import { Closer } from './Closer';
+import { Value } from './Value';
+import { Context } from './context';
+import { EventSender } from './events';
+
+export namespace Trackable {
+  export type Controller = Pick<EventSender, 'setContext' | 'sendEvent'>;
+  export type Cleanup = void | Closer;
+  export type Manager = (controller: Controller) => Cleanup;
+
+  class RevocableController implements Controller {
+    #delegate: () => Controller;
+
+    constructor(delegate: () => Controller) {
+      this.#delegate = delegate;
+    }
+    setContext(context: Context): void {
+      return this.#delegate().setContext(context);
+    }
+    sendEvent(name: string, message?: Value.Struct | undefined): void {
+      return this.#delegate().sendEvent(name, message);
+    }
+  }
+
+  export function setup(controller: Controller, manager: Manager): Closer {
+    let isClosed = false;
+    const delegate = () => {
+      if (isClosed) throw new Error('Controller is closed');
```

Comment:
Are we safe to throw an error here?

Reply:
I think we really should. It means a tracker is messing with the confidence instance after teardown. I.e. it didn't properly remove event handlers. It's a serious programming error.